### ISSUE TITLE
Make AI dumber for Instant battles option and AUTO mode

### ISSUE THREAD

**Issue by ihhub** [ihhub/fheroes2#6254]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.

### Describe the problem requiring a solution

I noticed that AI has become too smart and sometimes wins battles with even better results that an average human player. Instant battles are designed to help players in easy battles but not in complicated ones. The current state of the options leads to a situation that players don't even play battles and only focus on Adventure Map strategy. This shouldn't (my own opinion) happen in general as players suppose to play the full game.

### Describe the possible solution

Just make AI decisions in battle dumber. One way to do this is to always make AI to attack without defence actions. In this case AI might loose more army without trying to defend their units.

### Additional info

This topic open for discussions but I suggest to explain an easy to implement solution from development point of view.

**Comment by oleg-derevenetz** [comment]
I periodically watch HotA streams from pro gamers and even there, it is not so rare for AI to cope better than this pro player (he decides what he can do better, replays the battle and ends up with the worst losses), let alone an ordinary player. And no one raises the question about making the AI dumber :)

**Comment by Branikolog** [comment]
I agree with @ihhub . Autobattle AI shouldn't perform his best. I think, we shouldn't make him the dumbest player ever, but certainly now he plays better, than any novice player. So, how would new players study heroes battles then? :) Instant battle option is designed to work only for the cases where player's army is significantly stronger, than opponent's one.
Even in H3 autobattle AI behaves straightly: no waiting, no maneuvering. I think we should allow our instant AI to play similar. Without picking proper target (should attack the nearest one), always rushing ahead etc.

**Comment by Alucard648** [comment]
I wish that dumber insta-battles AI should be applicable only on higher difficulties. Maybe Rook or higher.

**Comment by ihhub** [comment]
Actually, @Alucard648 has a point. On higher difficulties it is reasonable to make AI dumber.

**Comment by zenseii** [comment]
Hi, everyone.

Am I misunderstanding or are we saying to make the AI progressively dumber the higher the difficulty is set to?

This seems contrary to what the opening message states. I think in a perfect world the instant battles would play exactly as you would. Since this is impossible I think the instant battles for easy difficulty should perform worse than for impossible difficulty. This should avoid novice players abusing it as a way to not gain combat skills.

**Comment by Alucard648** [comment]
> This seems contrary to what the opening message states. I think in a perfect world the instant battles would play exactly as you would. Since this is impossible I think the instant battles for easy difficulty should perform worse than for impossible difficulty. This should avoid novice players abusing it as a way to not gain combat skills.

Again, one can start auto-combat and learn certain battle skills as he can watch how AI performs the battle.

**Comment by zenseii** [comment]
Hi, @Alucard648.

The point is that a player won't do this because they will never see the combat screen with instant battles on. The times they do choose to restart they'll just realise that they always get a worse result so they might stick to always accepting the inatant result.

**Comment by idshibanov** [comment]
Is it really worth it? We can't police how people are playing the game. IMO if they want to do a quick run and focus on Adventure map - so be it.

When we get to implementing working multiplayer proper auto battles will be the key to save time (mostly early and midgame) and make it reasonable to finishing a game within few hours. If AI won't minimize losses on easy battles nobody will use it.

**Comment by oleg-derevenetz** [comment]
> If AI won't minimize losses on easy battles nobody will use it.

True. Too dumb AI is just useless for a relatively experienced player because instead of saving time, its use only gives unnecessary army losses. It also useless for a novice player, because, as @Alucard648 said, novice could at least learn from a strong AI by watching the battles in auto mode.

So, it is useless for an experienced player and is useless for a novice, what is the target audience of dumb AI then? With dumb AI, instant battles just will be always off, but they can be turned off in the same way now.

**Comment by LeHerosInconnu** [comment]
Hello everyone,

I think instant battle and auto battle should be as smart as possible, even if the result is better than a human player, otherwise why use them?
The player interested in combats will always try to do better, the player less interested in combats will not be "forced" to do them again, the experienced player will always want to do the most interesting and important combats himself.

**Comment by ihhub** [comment]
As I expected we have many different opinions. I am converting this into discussion.
